backend/src/main/scala/cromwell/backend/io/JobPaths.scala

```diff
@@ -10,26 +10,34 @@ object JobPaths {
   val CallPrefix = "call"
   val ShardPrefix = "shard"
   val AttemptPrefix = "attempt"
+  val CacheCopyPrefix = "cacheCopy"
   val ScriptPathKey = "script"
   val StdoutPathKey = "stdout"
   val StdErrPathKey = "stderr"
   val ReturnCodePathKey = "returnCode"
   val CallRootPathKey = "callRootPath"
   val DockerCidPathKey = "dockerCidPath"
 
-  def callPathBuilder(root: Path, jobKey: JobKey) = {
+  def callPathBuilder(root: Path, jobKey: JobKey, isCallCacheCopyAttempt: Boolean) = {
     val callName = jobKey.node.localName
     val call = s"$CallPrefix-$callName"
     val shard = jobKey.index map { s => s"$ShardPrefix-$s" } getOrElse ""
-    val retry = if (jobKey.attempt > 1) s"$AttemptPrefix-${jobKey.attempt}" else ""
 
-    List(call, shard, retry).foldLeft(root)((path, dir) => path.resolve(dir))
+    val retryOrCallCache =
+      if (isCallCacheCopyAttempt) CacheCopyPrefix
+      else if (jobKey.attempt > 1) s"$AttemptPrefix-${jobKey.attempt}"
+      else ""
+
+    List(call, shard, retryOrCallCache).foldLeft(root)((path, dir) => path.resolve(dir))
   }
 }
 
 trait JobPaths {
   import JobPaths._
 
+  def forCallCacheCopyAttempts: JobPaths
```

Comment:
Every time we override this method it becomes
```
override def forCallCacheCopyAttempts: JobPaths = this.copy(isCallCacheCopyAttempt = true)
```
Is there a reason we're not setting it as final here?

Reply:
Unfortunately there is... it's because `trait`s don't have a `copy` method.